src/components/CourseGradeItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Box, VStack, HStack, Text, Modal, Button, Input } from 'native-base';
+import { useState } from 'react';
+import { TouchableOpacity } from 'react-native';
+import { useDispatch } from 'react-redux';
+import { updateActualGrade } from '../redux/courseSlice';
+import * as SQLite from 'expo-sqlite';
+type CourseGradeItemProps = {
+  grade: {
+    id: number;
+    data: {
+      name: string;
+      actualScore: number | null;
+      maxScore: number;
+      weight: number;
+    };
+  };
+};
+
+const CourseGradeItem: React.FC<CourseGradeItemProps> = ({ grade }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [updatedActualGrade, setUpdatedActualGrade] = useState(grade.data.actualScore);
+  const dispatch = useDispatch();
+  const db = SQLite.openDatabase('db.gradetracker', '1.0');
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleSaveChanges = () => {
+    if (updatedActualGrade !== null) {
```

Comment:
Please also check if grade is greater than the max great, and show error in that case and not allow user to save.